src/pipeline.py

```diff
@@ -151,7 +163,13 @@ def build_pipeline(pipeline, source_config: SourceConfig, sink_config: SinkConfi
         | "ProcessOutput" >> beam.ParDo(PostProcessor())
     )
 
-    # save the predictions to a text file
-    predictions | "WriteOutputToGCS" >> beam.io.WriteToText(  # pylint: disable=expression-not-assigned
-        sink_config.output, shard_name_template="", append_trailing_newlines=True
-    )
+    # combine all the window results into one text for GCS
+    if source_config.streaming:
+        predictions | "WriteOutputToGCS" >> beam.io.fileio.WriteToFiles(  # pylint: disable=expression-not-assigned
```

Comment:
Can we simplify by just always using this setup?

Reply:
We could but I intend to show the difference here between these two write methods. Not a lot of people are aware about this.

Reply:
Ok, that's fine